### DISCUSSION TITLE
Normal users cannot load images normally

### DISCUSSION THREAD

**Discussion by miocoo** [requarks/wiki#3449] [Help / Questions]
**Question**
Normal users cannot load images normally

**Host Info (please complete the following information):**
OS: centos7, Docker
Wiki.js version: 2.5.144
Database engine: postgresq 12

Using the administrator to edit, the picture display is normal：
![image](https://user-images.githubusercontent.com/13830794/94355860-5ed70900-00ba-11eb-86b7-d9cc819be61a.png)

![image](https://user-images.githubusercontent.com/13830794/94354534-8b852380-00ae-11eb-8b10-5d3594bbfc5a.png)

However, after logging in with other users, the images cannot be displayed normally，the group to which the user belongs has been granted read permission：
![image](https://user-images.githubusercontent.com/13830794/94354499-35b07b80-00ae-11eb-84c2-2608114864f6.png)

**Comment by NGPixel** [comment]
But were read:assets permissions added to the group for the path where the asset is, not just the page?

If you upload an image to /some/path/image.jpg, then the group must have `read:assets` permission to the path `/some/path` as well.

**Comment by shegren** [comment]
Noticed this behavior when permissions (`read:pages` and `read:assets` in my case) are provided via `Tag Matches...`. With paths works fine, didn't test on regex though.

**Comment by rajid** [comment]
Running 2.5.287 and I'm still seeing this is a problem.  I can upload a file ("x.jpg") as the admin user but not as a normal user.  The normal user account has full permissions on these pages and can edit, etc., just fine.

The only error is "Error during upload" and a popup saying "File upload failed".  Both of these are totally useless errors an tell the user nothing at all other than the obvious.

So, there are 2 issues here:
1) the image upload fails for non-obvious reasons - not related to permissions or lack of sufficient disk space - while working fine for the admin account.
2) The error message is useless and should be changed to something useful.

I've been writing software for 40+ years and I have always made sure that error messages are useful.  This error message is a disgrace.

**Comment by rajid** [comment]
The answer, in my case, was that assets are stored, by default, at the root instead of under the path of the page itself.  My group didn't have permission to read and write assets at the root.  (It can be changed when storing the asset, but people don't notice that and there's no way to change the default location.)

The error message is still useless and a disgrace to the rest of the code!
